qualtran/bloqs/for_testing/costing.py

```diff
@@ -24,14 +24,23 @@ def _convert_callees(callees: Sequence[BloqCountT]) -> Tuple[BloqCountT, ...]:
     return tuple(callees)
 
 
+def _convert_static_costs(
+    static_costs: Sequence[Tuple[CostKey, Any]]
+) -> Tuple[Tuple[CostKey, Any], ...]:
+    # Convert to tuples in a type-checked way.
+    return tuple(static_costs)
+
+
 @frozen
 class CostingBloq(Bloq):
     """A bloq that lets you set the costs via attributes."""
 
     name: str
     num_qubits: int
     callees: Sequence[BloqCountT] = field(converter=_convert_callees, factory=tuple)
-    static_costs: Sequence[Tuple[CostKey, Any]] = field(converter=tuple, factory=tuple)
+    static_costs: Sequence[Tuple[CostKey, Any]] = field(
+        converter=_convert_static_costs, factory=tuple
```

Comment:
I feel like we have at least 2 of these _tuple attrs converters floating throughout the library. Should probably consolidate at some point.

Reply:
it's for the type checker. In theory we could re-use converters where the types are identical, but it's way more refactorable to keep the slight amount of duplication.

The type checker isn't smart enough if you use generic types in a universal to-tupler